core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -2422,7 +2428,11 @@ public PluginDowngradeJob(Plugin plugin, UpdateSite site, Authentication auth) {
 
         @Override
         protected URL getURL() throws MalformedURLException {
-            return new URL(plugin.url);
+            try {
+                return new URI(plugin.url).toURL();
+            } catch (MalformedURLException | URISyntaxException e) {
+                throw new RuntimeException(e);
```

Comment:
same as above - you should never be throwing RuntimeExeption where the method was previously docuemnted to throw a MalformedURLException

Reply:
Changed to
```
            try {
                return new URI(plugin.url).toURL();
            } catch (URISyntaxException e) {
                throw new MalformedURLException("Not a valid URL");
            }